### PR TITLE
Terraform infrastructure

### PR DESCRIPTION
**Pull request by s-olabina**
Created vpc, subnets, gateways, iam policy and ecr infrastructure with terraform.

### CODE REVIEW

File: infrastructure/modules/ECR/main.tf [hunk 1]
```diff
@@ -0,0 +1,28 @@
+module "ecr" {
+  source = "terraform-aws-modules/ecr/aws"
+
+  repository_name = "private-ecr-repo"
+
```
Comment (Igor-Knezevic-VegaIT) [inline infrastructure/modules/ECR/main.tf:5]: You are missing **repository_read_write_access_arns**

Comment (Igor-Knezevic-VegaIT) [reply infrastructure/modules/ECR/main.tf:5]: Value should be dynamically set from the iam role that you have made in different module.
P.s. Please don't resolve conversations, I will resolve them once I confirm that change is appropriate.

Comment (s-olabina) [reply infrastructure/modules/ECR/main.tf:5]: Isn't it dynamically created? I took the iam role from iam output file and then used it in root main file where I assign it to a variable. Then variable is assign to the repository_read_write_access_arns in ECR file.

Comment (Igor-Knezevic-VegaIT) [reply infrastructure/modules/ECR/main.tf:5]: Variable part should be dynamic, this one:

**var.repository_read_write_access_arns**

Instead you should call for module to look for the arn output of module that created iam role itself.

**Review by Igor-Knezevic-VegaIT — APPROVED** [review]

**Pull request merged**